src/js/tech/html5.js

```diff
@@ -86,6 +87,12 @@ class Html5 extends Tech {
    * @method dispose
    */
   dispose() {
+    let tt = this.el().textTracks;
+
+    tt.removeEventListener('change', Fn.bind(this, this.handleTextTrackChange));
```

Comment:
So, the issue seems that if we remove the event listener in the same event tick as removing the track, we don't actually get the `removetrack` event. Looking at the [spec](https://html.spec.whatwg.org/multipage/embedded-content.html#sourcing-out-of-band-text-tracks:event-media-removetrack), it seems like the `removetrack` events are queued to fire when tracks are removed, so, it makes sense for us to miss them.